src/JustSaying/Messaging/Channels/Receive/MessageReceiveBuffer.cs

```diff
@@ -154,6 +159,18 @@ await _sqsQueueReader
         return messages;
     }
 
+    private void CheckMessageReceiveControllerStatus(CancellationToken stoppingToken)
+    {
+        if (_messageReceiveController.ShouldStopReceiving)
+        {
+            _logger.LogInformation("Paused listening for messages from queue '{QueueName}'.", QueueName);
+
+            ((IMessageReceiveThreadPausingController)_messageReceiveController).PauseThreads(stoppingToken);
```

Comment:
We shouldn't do this - if someone tried to pass in a different implementation, they wouldn't be able to provide the internal interface on that type so this cast would fail.